src/FunctionApps/DSTP/src/main/kotlin/gov/cdc/prime/phdi/ParseHL7.kt

```diff
@@ -0,0 +1,66 @@
+package gov.cdc.prime.phdi
+
+import java.util.*
+import com.microsoft.azure.functions.*
+import com.microsoft.azure.functions.annotation.*
+
+import gov.cdc.prime.phdi.Utilities.HL7Reader
+import gov.cdc.prime.phdi.Utilities.HL7Parser
+import gov.cdc.prime.phdi.Utilities.HL7Validator
+
+class ParseHL7 {
+    @FunctionName("parsehl7")
+    fun run(
+        @BlobTrigger(
+            name = "file",
+            dataType = "binary",
+            path = "bronze/decrypted/{name}",
+            connection="AzureWebJobsStorage"
+        ) content: ByteArray,
+        @BindingName("name") filename: String,
+        @BlobOutput(
+            name="validTarget",
+            dataType = "string",
+            path="bronze/validation/{name}.valid",
+            connection="AzureWebJobsStorage"
+        ) validContent: OutputBinding<String>,
+        @BlobOutput(
+            name="invalidTarget",
+            dataType = "string",
+            path="bronze/validation/{name}.invalid",
+            connection="AzureWebJobsStorage"
+        ) invalidContent: OutputBinding<String>,
+        context: ExecutionContext
+	) {
+        context.logger.info("Name: ${filename} Size: ${content.size} bytes.")
```

Comment:
is the filename ever sensitive?

Reply:
I'm not sure, so to err on the side of caution, I've commented out the logger functionality. We can reassess at a later date.